oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -38,27 +41,50 @@
 @Slf4j
 public class StorageEsInstaller extends ModelInstaller {
     private final Gson gson = new Gson();
-
     private final StorageModuleElasticsearchConfig config;
     protected final ColumnTypeEsMapping columnTypeEsMapping;
 
+    /**
+     * The relation is between physical table and the specific structure definition.
+     */
+    private final ConcurrentHashMap<String, Map<String, Object>> tables;
+
     public StorageEsInstaller(Client client,
                               ModuleManager moduleManager,
-                              final StorageModuleElasticsearchConfig config) {
+                              StorageModuleElasticsearchConfig config) throws StorageException {
         super(client, moduleManager);
         this.columnTypeEsMapping = new ColumnTypeEsMapping();
         this.config = config;
+        this.tables = new ConcurrentHashMap<>(loadHistoryTables(client));
```

Comment:
I don't recommend to do this. Reading all template impact the user side **security policy**.
And according to existing codes, if this template and index exist, we will treat them as a correct version.

Reply:
You should read template settings when you face the creation event only.

Reply:
now we are facing more complex conditions. Since the template is created dynamically, we do not know when to terminate. It is not enough to judge by the existence of index and template.

Reply:
You need to check all, but don't scan all templates. This is not allowed due to security reason.